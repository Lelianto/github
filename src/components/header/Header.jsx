import React, { useState } from 'react';
import { slide as Menu } from 'react-burger-menu'
import { Link, useHistory } from 'react-router-dom'
import './Header.scss';
import { FaGithub, FaBell } from 'react-icons/fa';
import { Input } from 'element-react';

const Header = () => {
	let history = useHistory();
	const [text, setText] = useState('')

	const showSettings = (e) => {
		e.preventDefault();
	}

	const onChangeText = (e) => {
		setText(e)
	}

	const handleKeyPress = (e) => {
		if (e.which === 13) {
			history.push(`${text}`);
		}
	}

	return (
		<>
			<div className="bg-white w-full shadow-md">
				<div className="container w-full flex sm:justify-between md:justify-around">
					<div className="sm:block md:hidden mb-5">
						<Menu>
							<Link to={`/`} id="home" className="menu-item">Pull Request</Link>
							<Link to={`/`} onClick={showSettings} id="home" className="menu-item--small" >Issues</Link>
						</Menu>
					</div>
					<div className="sm:p-2">
						<FaGithub size={36} />
					</div>
					<div className="sm:hidden md:flex justify-left w-2/3" id="textbox">
						<Input onChange={(text) => onChangeText(text)} onKeyPress={(e) => handleKeyPress(e)} placeholder="Please input an username" className="max-w-lg h-4 relative my-auto mx-5" append="/" />
						<Link to={`/`} id="home" className="menu-item relative my-auto mx-5 text-gray-400 hover:text-black" >Pull Request</Link>
						<Link to={`/`} onClick={showSettings} id="home" className="menu-item--small relative my-auto mx-5 text-gray-400 hover:text-black" >Issues</Link>
					</div>
					<div className="sm:p-2 flex">
						<FaBell size={26} className="relative m-auto" />
					</div>
				</div>
			</div>
		</>
	)
}

export default Header