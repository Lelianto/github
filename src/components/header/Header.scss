#textbox .el-input-group {
  width: 35%;
  transition: width 0.5s;

  .el-input__inner {
	padding: 0px 10px;
    height: 30px;
  }
}

#textbox .el-input-group:focus-within {
  width: 100%;
  transition: width 0.5s;
}

@media only screen and (max-width: 768px) {
	/* Position and sizing of burger button */
	.bm-burger-button {
	  	position: relative;
	  	width: 32px;
	  	height: 24px;
		left: 15px;
    	top: 15px;
	}
	
	/* Color/shape of burger icon bars */
	.bm-burger-bars {
	  background: #000;
	}
	
	/* Color/shape of burger icon bars on hover*/
	.bm-burger-bars-hover {
	  background: #000;
	}
	
	/* Position and sizing of clickable cross button */
	.bm-cross-button {
	  height: 24px;
	  width: 24px;
	}
	
	/* Color/shape of close button cross */
	.bm-cross {
	  background: #bdc3c7;
	}
	
	/*
	Sidebar wrapper styles
	Note: Beware of modifying this element as it can break the animations - you should not need to touch it in most cases
	*/
	.bm-menu-wrap {
	  position: fixed;
	  width: 95%;
	  height: 100%;
	}
	
	/* General sidebar styles */
	.bm-menu {
	  background: #373a47;
	  padding: 2.5em 1.5em 0;
	  font-size: 1.15em;
	}
	
	/* Morph shape necessary with bubble or elastic */
	.bm-morph-shape {
	  fill: #373a47;
	}
	
	/* Wrapper for item list */
	.bm-item-list {
	  color: #b8b7ad;
	  padding: 0.8em;
	}
	
	/* Individual item */
	.bm-item {
	  display: inline-block;
	}
	
	/* Styling of overlay */
	.bm-overlay {
	  background: rgba(0, 0, 0, 0.3);
	}
}
