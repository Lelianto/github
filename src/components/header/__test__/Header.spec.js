import Header from "../Header";
import { shallow } from "enzyme";
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useParams: () => ({ username: 'Lelianto' }),
	useRouteMatch: () => ({ url: '/Lelianto' }),
	useLocation: () => ({ url: '/' })
}));

describe("Header: Check menu content", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Header lastUser={'last-user'} />)
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	it('Renders correctly', () => {
		const tree = renderer.create(<Router><Header lastUser={'last-user'} /></Router>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Pull request (1 and 2) menu is exist.", () => {
		const pullRequestMenu1 = wrapper.find('#pullrequest1');
		expect(pullRequestMenu1.text()).toEqual('Pull Request')
		const pullRequestMenu2 = wrapper.find('#pullrequest2');
		expect(pullRequestMenu2.text()).toEqual('Pull Request')
	});
	it("Issues (1 and 2) menu is exist.", () => {
		const issuesMenu1 = wrapper.find('#issues1');
		expect(issuesMenu1.text()).toEqual('Issues')
		const issuesMenu2 = wrapper.find('#issues2');
		expect(issuesMenu2.text()).toEqual('Issues')
	});
	it("Input Username.", () => {
		wrapper.find('#inputUsername').simulate('change', { target: { value: 'Lelianto' } });
		expect(wrapper.find("#inputUsername").get(0).props.value.target.value).toEqual("Lelianto");
	})
});