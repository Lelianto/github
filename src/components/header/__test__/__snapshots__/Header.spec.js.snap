// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: Check menu content Renders correctly 1`] = `
<div
  className="bg-white w-full shadow-md"
>
  <div
    className="sm:w-full md:container flex sm:justify-between md:justify-around"
  >
    <div
      className="sm:block md:hidden mb-5"
    >
      <div>
        <div
          className="bm-overlay"
          onClick={[Function]}
          style={
            Object {
              "MozTransform": "translate3d(100%, 0, 0)",
              "MsTransform": "translate3d(100%, 0, 0)",
              "OTransform": "translate3d(100%, 0, 0)",
              "WebkitTransform": "translate3d(100%, 0, 0)",
              "background": "rgba(0, 0, 0, 0.3)",
              "height": "100%",
              "opacity": 0,
              "position": "fixed",
              "transform": "translate3d(100%, 0, 0)",
              "transition": "opacity 0.3s, transform 0s 0.3s",
              "width": "100%",
              "zIndex": 1000,
            }
          }
        />
        <div
          style={Object {}}
        >
          <div
            className="bm-burger-button"
            style={
              Object {
                "zIndex": 1000,
              }
            }
          >
            <button
              id="react-burger-menu-btn"
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fontSize": 0,
                  "height": "100%",
                  "left": 0,
                  "margin": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
            >
              Open Menu
            </button>
            <span>
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "0%",
                  }
                }
              />
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "40%",
                  }
                }
              />
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "80%",
                  }
                }
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="bm-menu-wrap"
          id=""
          style={
            Object {
              "MozTransform": "translate3d(-100%, 0, 0)",
              "MsTransform": "translate3d(-100%, 0, 0)",
              "OTransform": "translate3d(-100%, 0, 0)",
              "WebkitTransform": "translate3d(-100%, 0, 0)",
              "height": "100%",
              "position": "fixed",
              "right": "inherit",
              "transform": "translate3d(-100%, 0, 0)",
              "transition": "all 0.5s",
              "width": "300px",
              "zIndex": 1100,
            }
          }
        >
          <div
            className="bm-menu"
            style={
              Object {
                "boxSizing": "border-box",
                "height": "100%",
                "overflow": "auto",
              }
            }
          >
            <nav
              className="bm-item-list"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <a
                className="bm-item menu-item"
                href="/last-user"
                id="pullrequest1"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                Pull Request
              </a>
              <a
                className="bm-item menu-item--small"
                href="/last-user"
                id="issues1"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                Issues
              </a>
            </nav>
          </div>
          <div
            style={Object {}}
          >
            <div
              className="bm-cross-button"
              style={
                Object {
                  "height": 24,
                  "position": "absolute",
                  "right": 8,
                  "top": 8,
                  "width": 24,
                }
              }
            >
              <button
                id="react-burger-cross-btn"
                onClick={[Function]}
                style={
                  Object {
                    "background": "transparent",
                    "border": "none",
                    "cursor": "pointer",
                    "fontSize": 0,
                    "height": "100%",
                    "left": 0,
                    "margin": 0,
                    "padding": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                tabIndex={-1}
              >
                Close Menu
              </button>
              <span
                style={
                  Object {
                    "position": "absolute",
                    "right": "14px",
                    "top": "6px",
                  }
                }
              >
                <span
                  className="bm-cross"
                  style={
                    Object {
                      "height": 14,
                      "position": "absolute",
                      "transform": "rotate(45deg)",
                      "width": 3,
                    }
                  }
                />
                <span
                  className="bm-cross"
                  style={
                    Object {
                      "height": 14,
                      "position": "absolute",
                      "transform": "rotate(-45deg)",
                      "width": 3,
                    }
                  }
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sm:p-2"
    >
      <a
        href="/last-user"
        id="logo"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={36}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 496 512"
          width={36}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          />
        </svg>
      </a>
    </div>
    <div
      className="sm:hidden md:flex justify-left w-2/3"
      id="textbox"
    >
      <div
        className="el-input el-input-group el-input-group--append max-w-lg h-4 relative my-auto mx-5"
        style={Object {}}
      >
        <input
          autoComplete="off"
          className="el-input__inner"
          id="inputUsername"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholder="Please input an username & press enter"
          type="text"
          value=""
        />
        <div
          className="el-input-group__append"
        >
          /
        </div>
      </div>
      <a
        className="menu-item relative my-auto mx-5 text-gray-400 hover:text-black"
        href="/last-user"
        id="pullrequest2"
        onClick={[Function]}
      >
        Pull Request
      </a>
      <a
        className="menu-item--small relative my-auto mx-5 text-gray-400 hover:text-black"
        href="/last-user"
        id="issues2"
        onClick={[Function]}
      >
        Issues
      </a>
    </div>
    <div
      className="sm:p-2 flex"
    >
      <svg
        className="relative m-auto"
        fill="currentColor"
        height={26}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width={26}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
        />
      </svg>
    </div>
  </div>
</div>
`;
